File: src/web/magmac/app/lang/node/CaseValues.ts
import { Lists } from "../../../../magmac/api/collect/list/Lists";
import { CompileResult } from "../../../../magmac/app/compile/error/CompileResult";
import { Node } from "../../../../magmac/app/compile/node/Node";
import { Deserializers } from "../../../../magmac/app/lang/Deserializers";
export class CaseValues {
	deserializeOrError(node : Node) : CompileResult<CaseValue> {return Deserializers.orError( "case-value", node, Lists.of( Deserializers.wrap( SingleCaseValue.deserialize), Deserializers.wrap( MultipleCaseValue.deserialize)));;}
}

File: src/java/magmac/app/lang/node/CaseValues.java
package magmac.app.lang.node;

/**
 * Utility class that deserializes the body of a {@code case} statement. It can
 * parse either a simple {@code case x;} or a block {@code case x { ... }}.
 */

import magmac.api.collect.list.Lists;
import magmac.app.compile.error.CompileResult;
import magmac.app.compile.node.Node;
import magmac.app.lang.Deserializers;

public final class CaseValues {
    public static CompileResult<CaseValue> deserializeOrError(Node node) {
        return Deserializers.orError("case-value", node, Lists.of(
                Deserializers.wrap(SingleCaseValue::deserialize),
                Deserializers.wrap(MultipleCaseValue::deserialize)
        ));
    }
}
